superset/views/database/views.py

```diff
@@ -92,21 +106,23 @@ def form_post(self, form):
             flash(message, "danger")
             return redirect("/csvtodatabaseview/form")
 
-        csv_file = form.csv_file.data
-        form.csv_file.data.filename = secure_filename(form.csv_file.data.filename)
         csv_filename = form.csv_file.data.filename
-        path = os.path.join(config["UPLOAD_FOLDER"], csv_filename)
+        extension = os.path.splitext(csv_filename)[1].lower()
+        path = tempfile.NamedTemporaryFile(
+            dir=app.config["UPLOAD_FOLDER"], suffix=extension
+        ).name
```

Comment:
Not sure if it makes any difference, but by default `NamedTemporaryFile` deletes the file after it is closed. I suppose it's ok to delete the file, too, but I think it should be safe to remove the `os.remove(path)`s below to keep the code lean.